### DISCUSSION TITLE
Windows 11 DX screen capture excesive contrast-wrong colors

### DISCUSSION THREAD

**Discussion by Pantera87** [awawa-dev/HyperHDR#924] [Q&A] [answered]
After a long time user of HyperHDR on tv, I decided to build another setup for my pc monitor (in SDR mode) using screen grabber. The problem I am faced with is that the screen capture is way too dark and colors are way off. I am attaching a screenshot for reference (look at the blue bar, it should be the same hue and brightness and still way off)

<img width="695" alt="Untitled" src="https://github.com/user-attachments/assets/b94902a7-169d-42e7-89df-7eec07204dda">

My setup is as follows: one physical 4k SDR monitor as well as a virtual display that is HDR.I thought the virtual display could play a role and I tried every way from disabling HDR on the virtual display all the way to completely uninstalling. The screen capture is selected on the SDR display.

I use two multisegment strips driven by an MH-ET live ESP32 running HyperSerialESP32

I tried also Hyperion to see if it is a widespread problem and the screen grabber there works properly, even though I couldn't get HyperSerial to connect, was giving an access denied error.

Another thing I messed with was the RGB threshold values on screen capture settings to no avail.

Any more ideas?

**Comment by awawa-dev** [comment]
What is your DX current capturing configuration: make a screenshot? Did you try to disable the hardware acceleration?

**Reply by Pantera87** [reply]
I tried playing with all settings including enabling/disabling hardware acceleration and manual signal detection with no results.

This is the current screen capture settings 

<img width="847" alt="Untitled" src="https://github.com/user-attachments/assets/dd6a1349-0f86-4da7-857c-8fd19fdd1a12">

**Reply by awawa-dev** [reply]
did you test older HyperHDR v19?

**Reply by Pantera87** [reply]
No I will do it now and report back the results. Could be the new pixel vertex and shaders of 20.0 be in play here?

**Reply by awawa-dev** [reply]
Make a conf backup first. Also this may be a key
![obraz](https://github.com/user-attachments/assets/cc055ecd-5a1c-44d6-8a89-9710a7cb5bf3)

**Reply by awawa-dev** [reply]
> Could be the new pixel vertex and shaders of 20.0 be in play here?

if you disable the hardware acceleration the pixel vertex&shaders are not used

**Reply by Pantera87** [reply]
Unfortunately still the same on 19.0 and on top of it I lost the esp handshake option making the esp work erratically (one segment is brighter than the other, I am not sure if it is related).

When disabling hardware acceleration does pc need reboot? When I change the setting there is no apparent change in the live preview. The same when changing HDR brightness correction

**Reply by awawa-dev** [reply]
no, the reset is not needed

**Reply by Pantera87** [reply]
Back to 20.0 with no change.I tried changing the HDR threshold value from 0 all the way to 10000 and still no change.

Is the dx11 grabber kernel based? I am not sure if the amd control panel could be interfering with the screen capture but playing around with their control panel I don't see any apparent setting that could interfere aside from live streaming that is disabled and never used

**Reply by awawa-dev** [reply]
This how SDR capturing works on my setup:
![obraz](https://github.com/user-attachments/assets/c3b4cc21-2acc-4096-a8b0-afa8030d16e5)

> Is the dx11 grabber kernel based? I am not sure if the amd control panel could be interfering with the screen capture but playing around with their control panel I don't see any apparent setting that could interfere aside from live streaming that is disabled and never used

try RDP to the PC and test the capturing then

**Reply by Pantera87** [reply]
I tried with RealVNC and still the same

Anyway I don't want to waste your valuable time, most propably it will be something interfering outside HyperHDR and corrupts the signal I have to eliminate one by one until I find the offender.

The strange thing is how screen capture in Hyperion works properly, isn't it the same as HyperHDR?

I think another major issue could be the VDD driver of the virtual display, most probably it confuses the SDR as HDR as there are mixed monitors one working in SDR and another one in HDR

Thank you very much for your time and assistance though! Keep up the good work with that excellent project

**Reply by awawa-dev** [reply]
RealVNC works in different way: your displays are still attached. RDP is using virtual display and should not be affected by AMD drivers. 
It is possible that some color profiles messed up with the capturing. It is a long shot bu some time ago missing this line after changes broke DirectX 11 capturing completely in v21: https://github.com/awawa-dev/HyperHDR/blob/313ca6216f427eee3b19e443d39bc9d4acc5fdab/sources/hyperhdr/main.cpp#L327

**Reply by Pantera87** [reply]
Thanks for the heads up, I have never used RDP before I will do it now and see if it makes a difference.

Another long shot could be windows color management, I am using DisplayCAL to force corrected lut on the screen but trying to bypass it completely didn't make any noticeable difference aside from the changing slightly the physical colors on the display (on live preview image is still super contrasty with crushed blacks)

Just to make sure is this line included in 20.0?If not I could give it a try, fork it and recompile it using this line if it will make any difference

**Reply by awawa-dev** [reply]
This line (completely directly unrelated to DirectX) was not needed in v20, but in v21 we dropped a lot of heavy QT GUI/widgets dependencies and as a surprising side-effect DirectX suddenly stopped working: https://github.com/awawa-dev/HyperHDR/pull/852#issuecomment-2116907968
If you want test v21 here are installers: https://github.com/awawa-dev/HyperHDR/actions/runs/10324710083 but I doubt it will change anything here.

**Reply by Pantera87** [reply]
I just installed 21.0 alpha and indeed made no difference

I tried changing the multimonitor setup from extending to only pc monitor and due to some glitch both screen and the live monitor became overexposed and super bright but they were in sync. Restarting chrome, both the tab and preview went as before (tab normal color preview crushed).I think the glitch was transfering the HDR option from second display to the main one that is not HDR compatible

I am getting the feeling that VDD is the main suspect but that is really strange as I have tried not only disabling but completely uninstalling it, except if it left some remnants behind that could interfere with the screen capture

**Reply by Pantera87** [reply, marked as answer]
Voila I found the answer!

As I was delving deeper into the settings for color management I came across a setting under display in windows settings called Automatically manage color for apps

<img width="761" alt="Untitled" src="https://github.com/user-attachments/assets/246f6e0b-fb21-4152-b286-659837e8dad7">

Changing it to off made the preview finally work as intended

Thanks for all your help!

**Reply by awawa-dev** [reply]
That's great! :) I looked for information about it and it seems that this option can generate a lot of problems with applications. It was recently also added to Win10. I couldn't find anything that would allow to block this option or how to bypass it in the API (Hyperion uses DX9 and not DX11 so that may explain the different behavior). I will add the information to the FAQ later. Thank you for your curiosity and for wanting to test different system options.
